### PR TITLE
Simple Bibtex citation update (Issue #3114)

### PR DESCRIPTION
**Pull request by bmorris3**
My earlier PR had patches for two issues (sorry about that!). I've separated them into their own branches, and I'm resubmitting them in separate PRs. 

This addresses Issue #3114.

### CODE REVIEW

File: astropy/__init__.py [hunk 1]
```diff
@@ -296,6 +296,19 @@ def _rebuild_extensions():
                       'with error code {0}: try rerunning this command '
                       'manually to check what the error was.'.format(
                           sp.returncode))
+# Set the bibtex entry to the article referenced in CITATION
+def _get_bibtex():
+    import os
+    import re
```
Comment (embray) [inline astropy/__init__.py]: These imports can just go at the top of the module.  I'm certain `os` at least has got to be already there:)
We also need to update the package setup to include the `CITATION` file in installs (and then get rid of the `if os.path.exists` clause).

**Comment by eteq** [comment]
For reference, the earlier issue was #3624 (it contains some useful discussion about this)

**Comment by eteq** [comment]
Also, this looks all good.  Thanks @bmorris3 - merging!

**Comment by embray** [comment]
I would like to put this in v1.0.3--although it is not a bug fix it is a harmless addition that would be good to have right away.

**Pull request merged**